docs/guides/migrating-from-camunda-platform-7/migration-readiness.md

```diff
@@ -0,0 +1,273 @@
+      }
+    }
+
+    String accumulatedRiskLevel = "green";
+    if (riskLevels.contains("rot") || riskLevels.contains("red")) {
+      accumulatedRiskLevel = "red";
+    } else if (riskLevels.contains("gelb") || riskLevels.contains("yellow")) {
+      accumulatedRiskLevel = "yellow";
+    }
+
+    execution.setVariable("risks", Variables.objectValue(risks).serializationDataFormat(SerializationDataFormats.JSON).create());
+    execution.setVariable("riskLevel", accumulatedRiskLevel);
+  }
+}
+```
+
+With FEEL, you can evaluate that data structure directly and have an expression on the "red" path:
+
+```
+= some risk in riskLevels satisfies risk = "red"
+```
+
+Additionally, you can even hook in FEEL as the scripting language in Camunda Platform 7 today (as explained by [Scripting with DMN inside BPMN](https://camunda.com/blog/2018/07/dmn-scripting/) or [User Task Assignment based on a DMN Decision Table](https://camunda.com/blog/2020/05/camunda-bpm-user-task-assignment-based-on-a-dmn-decision-table/)).
+
+However, more commonly you will keep using JUEL in Camunda Platform 7. If you write simple expressions, they can be easily migrated automatically, as you can see in [the test case](https://github.com/camunda-community-hub/camunda-7-to-8-migration/blob/main/modeler-plugin-7-to-8-converter/client/JuelToFeelConverter.test.js) of the [migration community extension](https://github.com/camunda-community-hub/camunda-7-to-8-migration). You should avoid more complex expressions if possible.
+
+Very often, a good workaround to achieve this is to adjust the output mapping of your Java Delegate to prepare data in a form that allows for easy expressions.
+
+Avoid hooking in Java code during an expression evaluation. The above listener to process the DMN result was one example of this, but a more diabolic example could be the following expression in Camunda Platform 7:
+
+```java
+// DON'T DO THIS:
+#{ dmnResultChecker.check( riskDMNresult ) }
+```
+
+Now, the dmnResultChecker is a Spring bean that can contain arbitrary Java logic, possibly even querying some remote service to query whether we currently accept yellow risks or not. Such code can not be executed within Camunda 8 FEEL expressions, and the logic needs to be moved elsewhere.
+
+### Camunda Forms
+
+Finally, while Camunda Platform 7 supports [different types of task forms](https://docs.camunda.org/manual/latest/user-guide/task-forms/), Camunda Platform 8 only supports [Camunda Forms](/guides/utilizing-forms.md) (and will actually be extended over time). If you rely on other form types, you either need to make Camunda Forms out of them or use a bespoke tasklist where you still support those forms.
+
+## Migration steps
+
+For migration, you need to look at development artifacts (BPMN models and application code), but also at workflow engine data (runtime and history) in case you migrate a process solution running in production.
+
+The typical steps are:
+
```

Comment:
Maybe as a first step: Analyze your current development artifacts with e.g. the diagram converter as it will give you a good overview of required steps. 
If you follow the migration steps linearly, you run from one issue into the next one. Maybe it's better to first have the complete picture of what needs to be done. I think we should highlight this here somehow, especially when splitting the guide into multiple pages

Reply:
My gut says we remove the numerics since that implies you move from one step to the next. I'll add a sentence about encountering issues along the way to paint a more realistic picture here.